calib_file.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+import pandas as pd
+
+class CalibFile:
+    '''
+    CalibFile stores calibration points.
+
+    The data is stored in a pandas data frame for easy r/w. 90% of this class
+    is just properties so the rows and columns can be r/w'ed in python as if 
+    they were just normal variables.
+
+    Properties
+    ----------
+
+    start_pt : pd.Series
+        Vector storing the location of the first calibration point. Can be set
+        with np array or pd series indexed ['x','y','z'].
+
+    n_pt : pd.Series
+        Vector storing the location of the calibration point on the N axis. Can
+        be set with np array or pd series indexed ['x','y','z'].
+
+
+    m_pt : pd.Series
+        Vector storing the location of the calibration point on the M axis. Can
+        be set with np array or pd series indexed ['x','y','z'].
+
+
+    N : int
+        number of samples traversed from start_pt to reach n_pt
+
+    M : int
+        number of samples traversed from start_pt to reach m_pt
+
+    '''
+    def __init__(self,from_file_name=None):
+        '''
+        Create an empty calibration instance or construct one from the given
+        file.
+        '''
+        self.axis = ['x','y','z']
+        if from_file_name != None:
+            self.data = self._read_from_file(from_file_name)
+        else:
+            start_pt_row = pd.Series([np.nan],index=self.axis)
+            n_pt_row = pd.Series([np.nan],index=self.axis + ['n'])
+            m_pt_row = pd.Series([np.nan],index=self.axis + ['m'])
+            self.data = pd.DataFrame(
+                [start_pt_row,n_pt_row,m_pt_row],
+                index = ['start_pt','n_pt','m_pt']
+            )
+    
+    def save(self,file_name):
+        '''
+        Store this calibration as a file
+        '''
+        self.data.to_csv(file_name)
+
+    def _read_from_file(self,file_name):
+        data = pd.read_csv(file_name,header=0,index_col=0)
+        return data
+
+    def load_file(self,file_name):
+        self.data = self._read_from_file(file_name)
+
+    def set_vector(self,row,vector):
+        self.data.loc[row,self.axis] = pd.Series(
+            vector,index=self.axis
+        )
+    
+    def get_vector(self,row):
+        return self.data.loc[row,self.axis]
+
+    def _get_N(self):
+        return self.data.loc['n_pt','n']
+        
+    def _set_N(self,value):
+        self.data.loc['n_pt','n'] = value
+    
+    N = property(_get_N,_set_N)
```

Comment:
There are builtin decorators that achieve this. Assuming you want to create an attribute named `N`:
```
@property
def N(self):
    return self.data.loc['n_pt', 'n']
@N.setter
def N(self, value):
    self.data.loc['n_pt', 'n'] = value
```
This is functionally equivalent to how you implemented it but more concise.